src/builder/build_traits.rs

```diff
@@ -61,11 +61,10 @@ pub trait Container {
     ///
     /// This function will return an error if there is an error in adding the
     /// [`OpType::Const`] node.
-    fn add_constant(&mut self, val: ConstValue) -> Result<ConstID, BuildError> {
-        let typ = val.const_type();
-        let const_n = self.add_child_op(ops::Const(val))?;
+    fn add_constant(&mut self, value: ConstValue, typ: ClassicType) -> Result<ConstID, BuildError> {
+        let const_n = self.add_child_op(ops::Const::new(value, typ))?;
```

Comment:
note `Const::new(value,typ)` returns a Result here so there should be a `?` inside the outer brackets